spotify_player/src/event/page.rs

```diff
@@ -90,38 +90,84 @@ fn handle_command_for_library_page(
 ) -> Result<bool> {
     if command == Command::Search {
         ui.new_search_popup();
-        Ok(true)
-    } else {
-        let data = state.data.read();
-        let (focus_state, folder_id) = match ui.current_page() {
-            PageState::Library { state } => (state.focus, state.playlist_folder_id),
-            _ => anyhow::bail!("expect a library page state"),
-        };
-        match focus_state {
-            LibraryFocusState::Playlists => Ok(window::handle_command_for_playlist_list_window(
+        return Ok(true);
+    }
+
+    let (focus_state, folder_id) = match ui.current_page() {
+        PageState::Library { state } => (state.focus, state.playlist_folder_id),
+        _ => anyhow::bail!("expect a library page state"),
+    };
+
+    if command == Command::SortLibraryAlphabetically {
+        let mut data = state.data.write();
+
+        // Sort playlists alphabetically
+        data.user_data.playlists.sort_by(|a, b| {
+            a.to_string().to_lowercase().cmp(&b.to_string().to_lowercase())
+        });
+
+        // Sort albums alphabetically
+        data.user_data.saved_albums.sort_by(|x, y| {
+            x.name.to_lowercase().cmp(&y.name.to_lowercase())
+        });
+
+        data.user_data
+            .persist_sorted_albums(&config::get_config().cache_folder)?;
+    }
+
+    if command == Command::SortLibraryByRecent {
+        let mut data = state.data.write();
+
+        // Sort playlists by ascending snapshot_id
+        data.user_data.playlists.sort_by(|a, b| {
+            match (a, b) {
+                (PlaylistFolderItem::Playlist(p1), PlaylistFolderItem::Playlist(p2)) => {
+                    p1.snapshot_id.cmp(&p2.snapshot_id) // Sort ascending by `snapshot_id`
```

Comment:
I think the correct ordering logic for playlists is comparing `current_folder_id` and if two playlists are in different folder, sort them accordingly, e.g

```rust
if p1.current_folder_id != p2. current_folder_id {
  p1.current_folder_id.cmp(&p2.current_folder_id)
} else {
  p1.snapshot_id.cmp(&p2.snapshot_id)
}
```